modules/camera/camera_device.py

```diff
@@ -8,24 +8,43 @@
 import cv2
 import numpy as np
 
+try:
+    import picamera2
+except ImportError:
+    pass
+
 
 class CameraDevice:
     """
     Wrapper for camera.
     """
 
-    def __init__(self, name: "int | str", save_nth_image: int = 0, save_name: str = "") -> None:
+    def __init__(
+        self, use_picamera: bool, name: "int | str", save_nth_image: int = 0, save_name: str = ""
+    ) -> None:
         """
-        name: Device name or index (e.g. /dev/video0 ).
+        use_pc2: Use picamera2 implementation instead of opencv impl
+        (optional) name: Device name or index (e.g. /dev/video0 ).
         (optional) save_nth_image: For debugging, saves every nth image.
             A value of 0 indicates no images should be saved
         (optional) save_name: For debugging, file name for saved images.
         """
-        self.__camera = cv2.VideoCapture(name)
-        assert self.__camera.isOpened()
-
-        self.__camera.set(cv2.CAP_PROP_FRAME_WIDTH, sys.maxsize)
-        self.__camera.set(cv2.CAP_PROP_FRAME_HEIGHT, sys.maxsize)
+        self.__using_picamera = use_picamera
+        if self.__using_picamera:
+            self.__camera = picamera2.Picamera2()
+            # maybe use create_still_configuration()
+            # if format is bad, use RGB 888 for [B, G, R] layout. BGR888 uses [R, G, B] layout
+            # see section 4.2.2.2 bottom warning for explanation
+            config = self.__camera.create_preview_configuration(
+                {"size": (1920, 1080), "format": "BGR888"}
```

Comment:
Instead of hardcoding this here, export it by including the image resolution in the __init__() parameters. Later, users can pass it in (in our case, we will use the values from the config file in cvp repo, when we call try to init this object later)